test/verify/check-openshift

```diff
@@ -476,6 +549,226 @@ LABEL io.projectatomic.nulecule.atomicappversion="0.1.11" \
         b.click("#vm-testvm-pod > a")
         b.wait_js_cond('window.location.hash === "#/l/pods/kube-system/%s"' % (pod_name))
 
+    def testKubevirtMachinesCreate(self):
+        self.bootstrapKubevirt()
+
+        runner = TestOpenshift.CreateVmRunner(self)
+
+        def checkDialogUiErrorTest(dialog, error_dict):
+            dialog.open() \
+                .fill() \
+                .createAndExpectUiError(error_dict) \
+                .cancel()
+            runner.checkEnvIsEmpty()
+
+        def checkDialogBackendErrorTest(dialog, error):
+            dialog.open() \
+                .fill() \
+                .createAndExpectBackendError(error) \
+                .cancel()
+            runner.checkEnvIsEmpty()
+
+        def createTest(dialog):
+            runner.tryCreate(dialog) \
+                .deleteVm(dialog) \
+                .checkEnvIsEmpty()
+
+        def createDuplicateTest(dialog):
+            runner.tryCreate(dialog)
+            dialog.open() \
+                .fill() \
+                .createAndExpectBackendError("already exists") \
+                .cancel()
+            runner.deleteVm(dialog) \
+                .checkEnvIsEmpty()
+
+        b = self.browser
+        self.login_and_go("/kubernetes"
+                          )
+        # kubevirt is available, so link should be present
+        b.wait_present("#vms-menu-link")
+        b.click("#vms-menu-link")
+        runner.checkEnvIsEmpty()
+
+        checkDialogUiErrorTest(
+            TestOpenshift.VmDialog(self, 'testvm1', ''),
+            {'#resource-text-error': "VM definition is required"}
+        )
+
+        checkDialogUiErrorTest(
+            TestOpenshift.VmDialog(self,'testvm2', '{a": "b"}'),
+            {'#resource-text-error': "VM definition is not a valid JSON"}
+        )
+
+        checkDialogUiErrorTest(
+            TestOpenshift.VmDialog(self,'testvm3', '[{}]'),
+            {'#resource-text-error': "VM definition should be an object. Not an array"}
+        )
+
+        checkDialogBackendErrorTest(
+            TestOpenshift.VmDialog(self, 'testvm4', '{}'),
+            "Failure"
+        )
+
+        vm_name = 'testvm5'
+        createTest(
+            TestOpenshift.VmDialog(self, vm_name, TEST_VM_JSON % vm_name)
+        )
+
+        vm_name = 'testvm6'
+        createTest(
+            TestOpenshift.VmDialog(self, vm_name, TEST_VM_JSON % vm_name, namespace='default')
+        )
+
+        vm_name = 'testvm7'
+        createTest(
+            TestOpenshift.VmDialog(self, vm_name, TEST_VM_JSON % vm_name, namespace='kube-public')
+        )
+
+        vm_name = 'testvm8'
+        createTest(
+            TestOpenshift.VmDialog(self, vm_name, TEST_VM_JSON % vm_name, namespace='kube-system')
+        )
+
+        vm_name = 'testvm9'
+        createDuplicateTest(
+            TestOpenshift.VmDialog(self, vm_name, TEST_VM_JSON % vm_name)
+        )
+
+    class VmDialog:
```

Comment:
Putting a `class` into a `class` in between method calls (and below where it gets used) is strange. Please move this into `testKubevirtMachinesCreate()` as a method-private class, or at the top of the file if you want to reuse it.

Reply:
Ok I will put it into the function which is not much readable IMO. Anyway I would prefer if we could have source code directory for each `check-.*` with multiple files. It would be easier to manage and follow.